packages/clarity-js/src/data/metadata.ts

```diff
@@ -88,9 +88,9 @@ export function start(): void {
   // If consent status is not already set, initialize it based on project configuration. Otherwise, use the existing consent status.
   if (consentStatus === null) {
     consentStatus = {
-      source: ConsentSource.Implicit,
-      ad_Storage: config.track ? Constant.Granted : Constant.Denied,
-      analytics_Storage: config.track ? Constant.Granted : Constant.Denied,
+      source: u.consent || u.ad_consent ? ConsentSource.Cookie : ConsentSource.Implicit,
+      ad_Storage: u.ad_consent || (config.track && consentStatus.source != ConsentSource.Cookie) ? Constant.Granted : Constant.Denied,
```

Comment:
why the additional check for source? we shouldnt be checking config.track with this approach. There should be a ad and analytics value check instead